sds_data_manager/lambda_code/SDSCode/indexer.py

```diff
@@ -47,17 +51,52 @@ def lambda_handler(event, context):
         filename = record["s3"]["object"]["key"]
 
         logger.info(f"Attempting to insert {os.path.basename(filename)} into database")
-        # TODO: change below logics to use new FilenameParser
-        # when we create schema and write file metadata to DB
         filename_parsed = FilenameParser(filename)
-        filename_parsed.upload_filepath()
-        metadata = None
+        filepath = filename_parsed.upload_filepath()
 
-        # TODO: remove this check since upload api validates filename?
-        # Found nothing. This should probably send out an error notification
-        # to the team, because how did it make its way onto the SDS?
-        if metadata is None:
-            logger.info("Found no matching file types to index this file against.")
-            return None
+        # confirm that the file is valid
+        if filepath["statusCode"] == 400:
+            logger.error(filepath["body"])
+            break
 
-        logger.info("Found the following metadata to index: " + str(metadata))
+        # setup a dictionary of metadata parameters to unpack in the
+        # instrument table
+        metadata_params = {
+            "file_path": filepath["body"],
+            "instrument": filename_parsed.instrument,
+            "data_level": filename_parsed.data_level,
+            "descriptor": filename_parsed.descriptor,
+            "start_date": filename_parsed.startdate,
+            "end_date": filename_parsed.enddate,
+            "ingestion_date": datetime.datetime.now(),
+            "version": filename_parsed.version,
+            "extension": filename_parsed.extension,
+        }
+
+        # Add data to the corresponding instrument database
+        with Session(engine) as session:
+            if filename_parsed.instrument == "lo":
+                data = models.LoTable(**metadata_params)
```

Comment:
I'd write this as a lookup table outside of the session if you're not relying on the session for this portion.

```python
model_lookup = {"lo": models.LoTable, "hi": models.HiTable}
data = model_lookup[filename_parsed.instrument](**metadata_params)
with Session(engine) as session:
    session.add(data)
    session.commit()
```

Reply:
nice!